src/Nerdbank.Streams.Tests/NestedStreamTests.cs

```diff
@@ -90,13 +113,82 @@ public void SetLength()
     }
 
     [Fact]
-    public void Seek()
+    public void Seek_Current()
     {
-        Assert.Throws<NotSupportedException>(() => this.stream.Seek(0, SeekOrigin.Begin));
+        Assert.Equal(0, this.stream.Position);
+        Assert.Equal(0, this.stream.Seek(0, SeekOrigin.Current));
+        Assert.Equal(0, this.underlyingStream.Position);
+        Assert.Throws<IOException>(() => this.stream.Seek(-1, SeekOrigin.Current));
+        Assert.Equal(0, this.underlyingStream.Position);
+
+        Assert.Equal(5, this.stream.Seek(5, SeekOrigin.Current));
+        Assert.Equal(5, this.underlyingStream.Position);
+        Assert.Equal(5, this.stream.Seek(0, SeekOrigin.Current));
+        Assert.Equal(5, this.underlyingStream.Position);
+        Assert.Equal(4, this.stream.Seek(-1, SeekOrigin.Current));
+        Assert.Equal(4, this.underlyingStream.Position);
+        Assert.Throws<IOException>(() => this.stream.Seek(-10, SeekOrigin.Current));
+        Assert.Equal(4, this.underlyingStream.Position);
+
+        Assert.Equal(0, this.stream.Seek(0, SeekOrigin.Begin));
+        Assert.Equal(0, this.stream.Position);
+
+        Assert.Equal(DefaultNestedLength + 1, this.stream.Seek(DefaultNestedLength + 1, SeekOrigin.Current));
+        Assert.Equal(DefaultNestedLength + 1, this.underlyingStream.Position);
+        Assert.Equal((2 * DefaultNestedLength) + 1, this.stream.Seek(DefaultNestedLength, SeekOrigin.Current));
+        Assert.Equal((2 * DefaultNestedLength) + 1, this.underlyingStream.Position);
```

Comment:
Looking at the underlying stream's `Position` is interesting. I may add a test where `ReadSlice` isn't the first API called on the underlying stream so we can test how `Position` behaves when it *doesn't* equal the underlying stream's position.